command/meta.go

```diff
@@ -212,11 +226,96 @@ func (m *ApiMeta) initializeX509Config(client http.Client, clientCA []byte, cert
 	client.Transport.(*http.Transport).TLSClientConfig.MinVersion = tls.VersionTLS12
 	client.Transport.(*http.Transport).TLSClientConfig.PreferServerCipherSuites = true
 	client.Transport.(*http.Transport).TLSClientConfig.Certificates = []tls.Certificate{cert}
-	client.Transport.(*http.Transport).TLSClientConfig.InsecureSkipVerify = true // TODO(jacobkiefer): Add a flag this.
+	client.Transport.(*http.Transport).TLSClientConfig.InsecureSkipVerify = true // TODO(jacobkiefer): Add a flag for this.
 
 	return &client
 }
 
+func (m *ApiMeta) Authenticate() error {
+	auth := m.Config.Auth
+	if auth != nil && auth.Enabled && auth.OAuth2 != nil {
+		OAuth2 := auth.OAuth2
+		if !OAuth2.IsValid() {
+			return errors.New("Incorrect OAuth2 auth configuration.")
+		}
+
+		usr, err := user.Current()
+		if err != nil {
+			m.Ui.Error("Could not read current user from environment, failing.")
+			return err
+		}
+		credentialsLocation := filepath.Join(usr.HomeDir, ".spin", "credentials")
```

Comment:
Why not write the token into the ~/.spin/config file?

Reply:
No reason I guess. I imagined users possibly wanting to delete `credentials` to periodically refresh, but there's no reason they couldn't edit the config file either. The stored token is json for the refresh token and expiry information, so I wonder if that will work nicely with yaml - I'll give it a shot.